.travis.yml

```diff
@@ -3,19 +3,52 @@ os:
   - linux
   - osx
 solution: MoreLinq.sln
-mono: latest
+mono: 4.8.0
 dist: trusty
 sudo: required
-dotnet: 1.0.0-preview2-003131
+dotnet: 1.0.1
 env:
   - CONFIGURATION=Debug
   - CONFIGURATION=Release
 
+# Ensure MSBuild is installed
+before_install:
+  - | 
+    if [ "$TRAVIS_OS_NAME" == "linux" ]; then
+      sudo apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys 3FA7E0328081BFF6A14DA29AA6A19B38D3D831EF
+      echo "deb http://download.mono-project.com/repo/debian wheezy main" | sudo tee /etc/apt/sources.list.d/mono-xamarin.list
+      sudo apt-get -qq update
+      sudo apt-get install msbuild
+    fi
+
+  - |
+    # This is a temporary fix to workaround a problem where Travis picks up a build of Mono on OSX that
+    # includes a broken version of MSBuild.
+    #
+    # See https://github.com/mono/msbuild/commit/cff4013ba3a69f82dc0ae96b3e15af700d8f74ef
+    # for the fix this is replicating.
+    if [ "$TRAVIS_OS_NAME" == "osx" ]; then
+      MSBUILD_BIN=/Library/Frameworks/Mono.framework/Versions/4.8.0/lib/mono/msbuild/15.0/bin
+
+      if [ ! -d $MSBUILD_BIN ] || [ -f $MSBUILD_BIN/System.Reflection.Metadata.dll ]; then
+        echo "WORKAROUND: Time to remove System.Reflection.Metadata.dll workaround"
+      else
+        echo "WORKAROUND: Copying System.Reflection.Metadata.dll to Mono MSBuild"
+        sudo cp $MSBUILD_BIN/Roslyn/System.Reflection.Metadata.dll $MSBUILD_BIN/System.Reflection.Metadata.dll
+      fi
+    fi
+
+  - msbuild /version
+  - |
+  # Handle too many files on OS X
```

Comment:
Indentation is off here, and it causes yaml parsing to fail:

```
syntax error: (<unknown>): did not find expected '-' indicator while parsing a block collection at line 16 column 3
```

https://lint.travis-ci.org/